packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,294 @@
+                        );
+                break;
+            default:
+                if (construct.parameters.every(isParameterProperty)) return;
+
+                const allPropsCanBeParamProps = construct.parameters
+                    .filter((param) => !isParameterProperty(param))
+                    .every((param) => canBeParameterProperty(param, construct));
+
+                if (!allPropsCanBeParamProps && construct.parameters.some(isParameterProperty)) {
+                    for (const param of construct.parameters.filter(isParameterProperty))
+                        this.addFailure(
+                            param.pos,
+                            param.end,
+                            FAILURE_STRINGS[Mode.Consistent].cannotBeParamPropsOnly,
+                            getFixerForDisallowedParameterProp(
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+                } else if (allPropsCanBeParamProps && !construct.parameters.every(isParameterProperty)) {
+                    for (const param of construct.parameters.filter((p) => !isParameterProperty(p)))
+                        this.addFailure(
+                            param.pos,
+                            param.end,
+                            FAILURE_STRINGS[Mode.Consistent].canBeParamPropsOnly,
+                            getFixerForLonghandProp(param, construct),
+                        );
+                }
+        }
+    }
+}
+
+function canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
+    if (!isIdentifier(param.name) || param.dotDotDotToken) return false;
+    const member = construct.parent!.members.find((mem) => isPropertyMatchForParam(mem, param));
+    if (member && !member.decorators) {
+        for (const stmt of statementsMinusDirectivesAndSuper(construct)) {
+            if (isStatementWithPossibleSideEffects(stmt, param)) return false;
+            if (isSimpleParamToPropAssignment(stmt, param)) return true;
+        }
+    }
+
+    return false;
+}
+
+function getFixerForDisallowedParameterProp(
+    construct: ts.ConstructorDeclaration,
+    param: ts.ParameterDeclaration,
+    lineBreak: string,
+): Replacement[] {
+    return [
+        /* Add assignment after directives and super() call (if they exists) */
+        Replacement.append(
+            getEndOfDirectivesAndSuper(construct),
+            `${lineBreak}this.${param.name.getText()} = ${param.name.getText()};`,
+        ),
+
+        /* Add property to class */
+        Replacement.append(construct.getStart(), getPropertyTextFromParam(param) + lineBreak),
+
+        /* Finally, delete modifiers from the parameter prop */
+        Replacement.delete(param.decorators ? param.modifiers![0].getStart() : param.pos, param.name.getStart()),
```

Comment:
no need to check for decorators here, you can always use `param.modifiers!.pos` and `param.modifiers!.end` as replacement span